src/LSST/control/ControlMain.h

```diff
@@ -47,20 +56,52 @@ class ControlMain {
     /// Verify threads are stopped and joined.
     ~ControlMain();
 
-    /// Return a reference to the global ControlMain instance.
-    /// @throws `ConfigException` if `setup` has not already been called.
-    static ControlMain& get();
-
     /// Return a shared pointer to the global ControlMain instance.
     /// @throws `ConfigException` if `setup` has not already been called.
     static Ptr getPtr();
 
+    /// &&& doc
+    void run(int argc, const char* argv[]);
+
+    /// Returns true once the server is running.
+    bool getRunning() { return _running; }
+
+    /// Return a pointer to `_simCore`.
+    std::shared_ptr<simulator::SimCore> getSimCore() { return _simCore; }
+
+
+    /// &&& doc
+    void stop();
+
+    /// &&& doc
+    void join();
+
+    /// &&& doc
+    std::shared_ptr<system::ComControlServer> getComServer() {
```

Comment:
No single client in this branch uses this method. The direct access to the internal state of an object is a bad idea anyway.